charts/vector/templates/haproxy/_helpers.tpl

```diff
@@ -14,7 +14,7 @@ Common labels
 helm.sh/chart: {{ include "vector.chart" . }}
 {{ include "haproxy.selectorLabels" . }}
 {{- if .Chart.AppVersion }}
-app.kubernetes.io/version: {{ .Chart.AppVersion | quote }}
+app.kubernetes.io/version: {{ .Values.image.tag | default .Chart.AppVersion }}
```

Comment:
It's odd that this isn't the HAProxy version... but perhaps I intentionally did that for a reason I no longer remember. What do you think?